datafusion/common/src/config.rs

```diff
@@ -188,6 +195,211 @@ macro_rules! config_namespace {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct EncryptionKeyAsHex {
+    pub key: String,
+    pub key_metadata: Option<String>,
+}
+
+#[derive(Debug)]
+pub struct EncryptionKey {
+    pub key: Vec<u8>,
+    pub key_metadata: Option<Vec<u8>>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct EncryptionColumnKeys {
+    pub column_keys_as_hex: HashMap<String, EncryptionKeyAsHex>,
+}
+
+impl EncryptionColumnKeys {
+    pub fn new(
+        column_names: &Vec<String>,
+        keys: &Vec<Vec<u8>>,
+        meta: &Vec<Vec<u8>>,
+    ) -> Self {
+        let mut column_keys_as_hex: HashMap<String, EncryptionKeyAsHex> = HashMap::new();
+
+        for (i, column_name) in column_names.iter().enumerate() {
+            let key_metadata: Option<String> = meta.get(i).map(|x| hex::encode(x));
+            let encryption_key = EncryptionKeyAsHex {
+                key: hex::encode(keys[i].clone()),
+                key_metadata,
+            };
+            column_keys_as_hex.insert(column_name.clone(), encryption_key);
+        }
+
+        EncryptionColumnKeys { column_keys_as_hex }
+    }
+
+    pub fn to_json(&self) -> String {
+        serde_json::to_string(self).unwrap()
+    }
+
+    pub fn from_json_to_column_keys(s: &String) -> HashMap<String, EncryptionKey> {
+        let eck: EncryptionColumnKeys =
+            serde_json::from_str(s).expect("failed to decode column keys from JSON");
+        let mut hm: HashMap<String, EncryptionKey> = HashMap::new();
+        for (key, val) in eck.column_keys_as_hex {
+            let col = key;
+            let key = hex::decode(val.key).expect("Invalid column key");
+            let key_metadata = val
+                .key_metadata
+                .map(|x| hex::decode(x).expect("Invalid column metadata"));
+            let encryption_key = EncryptionKey { key, key_metadata };
+            hm.insert(col, encryption_key);
+        }
+        hm
+    }
+}
+
+config_namespace! {
+    pub struct ConfigFileEncryptionProperties {
+        pub encrypt_footer: bool, default = false
+        pub footer_key_as_hex: String, default = String::new()
+        pub footer_key_metadata_as_hex: String, default = String::new()
+        pub column_keys_as_json_hex: String, default = String::new()
+        pub aad_prefix_as_hex: String, default = String::new()
+        pub store_aad_prefix: bool, default = false
+    }
+}
+
+config_namespace! {
+    pub struct ConfigFileDecryptionProperties {
+        pub footer_key_as_hex: String, default = String::new()
+        pub column_keys_as_json_hex: String, default = String::new()
+        pub aad_prefix_as_hex: String, default = String::new()
+        pub footer_signature_verification: bool, default = true
+    }
+}
+
+#[cfg(feature = "parquet")]
+impl Into<FileDecryptionProperties> for ConfigFileDecryptionProperties {
```

Comment:
Should this implement `TryInto` instead so that we can return an error rather than panic when there's a misconfiguration?

Reply:
I feel like a panic is appropriate here. This shouldn't error unless there has been some terrible misconfiguration at which point it is better to panic and pinpoint there error rather than pass errors up the stack which makes the source harder to find.